src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -956,6 +930,35 @@ static void checkAndReportManagedTypes(TypeSymbol type, SyntaxNode syntax, bool
                         throw ExceptionUtilities.UnexpectedValue(type.ManagedKindNoUseSiteDiagnostics);
                 }
             }
+
+            static UnmanagedCallersOnlyAttributeData DecodeUnmanagedCallersOnlyAttributeData(SourceMethodSymbolWithAttributes @this, CSharpAttributeData attribute, Location location, DiagnosticBag diagnostics)
+            {
+                Debug.Assert(attribute.AttributeClass is not null);
+                ImmutableHashSet<CodeAnalysis.Symbols.INamedTypeSymbolInternal>? callingConventionTypes = null;
+                if (attribute.CommonNamedArguments is { IsDefaultOrEmpty: false } namedArgs)
+                {
+                    var systemType = @this.DeclaringCompilation.GetWellKnownType(WellKnownType.System_Type);
```

Comment:
This is mostly a copy/paste to move the function from `MethodSymbol` to where it actually matters. The only addition is the type check below, and retrieving `System.Type` here.